### PR TITLE
feat(codeowners): External User/Team mappings UI

### PR DESCRIPTION
**Pull request by NisanthanNanthakumar**
## Objective:
Implements the List view UI for External User/Team Mappings for integrations. Implements add/edit/delete functionality.

## UI:
Screen Shot 2021-04-16 at 7 32 03 PM
Screen Shot 2021-04-16 at 7 32 45 PM
Screen Shot 2021-04-16 at 7 32 33 PM
Screen Shot 2021-04-16 at 7 35 04 PM

## Tests:
Added acceptance tests with visual snapshots for relevant states.

### CODE REVIEW

**Comment by MeredithAnya** [comment]
@NisanthanNanthakumar I think if we are going to required the `@` to part of the external user or team we could add some text to the input or at least have the placeholder be `@org/teamname` or something, I don't think it's going to be obvious what they need to put

File: static/app/views/settings/organizationIntegrations/configureIntegration.tsx [hunk 1]
```diff
@@ -218,6 +220,8 @@ class ConfigureIntegration extends AsyncView<Props, State> {
     const tabs = [
       ['repos', t('Repositories')],
       ['codeMappings', t('Code Mappings')],
+      ['userMappings', t('User Mappings')],
```
Comment (MeredithAnya) [inline static/app/views/settings/organizationIntegrations/configureIntegration.tsx:223]: I think this is going to show the user mappings and team mappings to people who have stack trace linking but may not have codeowners right? seems like we need an extra check here

File: static/app/views/organizationIntegrations/integrationExternalTeamMappings.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          />
+        </Body>
+      </React.Fragment>
+    ));
+  };
+
+  renderBody() {
+    const {integration} = this.props;
+    return (
+      <React.Fragment>
```
Comment (scefali) [inline static/app/views/organizationIntegrations/integrationExternalTeamMappings.tsx:137]: @NisanthanNanthakumar Nit: do you need `<React.Fragment>` here?

File: static/app/components/integrationExternalMappingForm.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  render() {
+    const {onSubmitSuccess, onCancel, mapping, baseEndpoint, onSubmit} = this.props;
+
+    // endpoint changes if we are making a new row or updating an existing one
+    const endpoint = !baseEndpoint
+      ? undefined
+      : mapping
+      ? `${baseEndpoint}${mapping.id}/`
+      : baseEndpoint;
+    const apiMethod = !baseEndpoint ? undefined : mapping ? 'PUT' : 'POST';
```
Comment (scefali) [inline static/app/components/integrationExternalMappingForm.tsx:83]: @NisanthanNanthakumar a bit confused why we'd want `endpoint` and/or `apiMethod` to be `undefined`.

File: static/app/types/index.tsx [hunk 4]
```diff
@@ -2025,3 +2026,25 @@ export type KeyValueListData = {
   subjectDataTestId?: string;
   subjectIcon?: React.ReactNode;
 }[];
+
+export type ExternalActorMapping = {
+  id: string;
+  externalName: string;
+  memberId?: string;
+  teamId?: string;
+  sentryName: string;
```
Comment (scefali) [inline static/app/types/index.tsx:2035]: @NisanthanNanthakumar Wondering if `ExternalActorMapping` should be a union between two types: one with `memberId` and one with `teamId`. Is it possible `teamId` and `memberId` could be both missing or present?

File: static/app/components/integrationExternalMappingForm.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+        deprecatedSelectControl: false,
+      });
+    }
+    if (type === 'team') {
+      fields.push({
+        name: 'teamId',
+        type: 'select',
+        required: true,
+        label: tct('Sentry [type]', {type: capitalize(type)}),
+        placeholder: t(`Sentry ${capitalize(type)}`),
```
Comment (scefali) [inline static/app/components/integrationExternalMappingForm.tsx:66]: @NisanthanNanthakumar this isn't going to translate properly, you should use `tct` like you do with the label

File: static/app/components/integrationExternalMappingForm.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+import {FieldFromConfig} from 'app/views/settings/components/forms';
+import Form from 'app/views/settings/components/forms/form';
+import {Field} from 'app/views/settings/components/forms/type';
+
+type Props = {
+  organization: Organization;
+  integration: Integration;
+  onSubmitSuccess: Form['props']['onSubmitSuccess'];
+  onCancel: Form['props']['onCancel'];
+  onSubmit?: Form['props']['onSubmit'];
```
Comment (scefali) [inline static/app/components/integrationExternalMappingForm.tsx:16]: @NisanthanNanthakumar Nit: You could do something like this:
```
type Props = Pick<Form['props'], 'onSubmitSuccess' | 'onCancel' | 'onSubmit'> & {
```

File: static/app/types/index.tsx (hunk 4 above)
Comment (NisanthanNanthakumar) [reply static/app/types/index.tsx:2035]: @scefali no, it's either or. I think the current type is ok. We can refactor in the future if this type becomes unwieldily.

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**